dist/sgx/scripts/install_service

```diff
@@ -1,67 +1,197 @@
+    if [ -d "$INSTALL_DIR" ]; then
+        rmdir $INSTALL_DIR
+    fi
+}
+
+function error() {
+    if [ $? -ne 0 ]; then
+        cleanUp
+        print_error $1
+        exit 1
+    fi
+}
+
+function checkHsmBinaries() {
+    # Check for HSM binary files
+    FILES="$HSMBIN_DIR/hsmsgx $HSMBIN_DIR/hsmsgx_enclave.signed"
+    for f in ${FILES}; do
+        if [[ ! -e $f ]]; then
+            print_error "HSM binary file '$(basename $f)' does not exist."
+            exit 1
+        fi
+    done
+}
+
+function failIfServiceExists() {
+    # Check if a service with the same name is currently installed
+    if systemctl list-units --full --all | grep -Fq $SERVICE_NAME; then
+        print_error "Service $SERVICE_NAME is already installed. Please uninstall it first."
+        exit 1
+    fi
+}
+
+function selectInstallationDir() {
+    while true; do
+        print_info "Enter the absolute path to the installation directory (empty directory name to abort)"
+        while true; do
+            read -p "> " INSTALL_DIR
+            if [[ "$INSTALL_DIR" == "" ]]; then
+                print_error "Aborted."
+                exit 1
+            fi
+            INSTALL_DIR=$(realpath $INSTALL_DIR)
+
+            if [ -d "$INSTALL_DIR" ]; then
+                print_error "Directory exists: $INSTALL_DIR. Try again or enter empty directory name to abort."
+            else
+                if ! mkdir "$INSTALL_DIR"; then
+                    print_error "Could not create directory: $INSTALL_DIR. Try again or enter empty directory name to abort."
+                else
+                    print_warning "powHSM will be installed to $INSTALL_DIR"
+                    break
+                fi
+            fi
+        done
+        print_info "Proceed? [Y/N]"
+        read -p "> " proceed
+        if [[ "Y" == "$proceed" ]] || [[ "y" == "$proceed" ]]; then
+            break;
+        else
+            rmdir $INSTALL_DIR
+        fi
+    done
+}
+
+function createServiceUnit() {
+    rm -rf $SERVICE_DIR
+    mkdir $SERVICE_DIR
+
+    cp $SERVICE_UNIT $SERVICE_DIR
+    # Replace the $HSM_INSTALL_DIR token in the script with the actual installation directory
+    sed -i "s|\$HSM_INSTALL_DIR|$INSTALL_DIR|g" $SERVICE_DIR/$SERVICE_NAME
+    error "Could not replace the installation directory in the service unit file."
+}
+
+function installPowHsm() {
+    mkdir $INSTALL_DIR/bin
+    error "Could not create the installation directory."
+    cp -R $HSMBIN_DIR/* $INSTALL_DIR/bin
+    error "Could not copy the powHSM binaries to the installation directory."
+    cp -R $ROOT_DIR/Dockerfile $INSTALL_DIR/bin
+    error "Could not copy the Dockerfile to the installation directory."
+    cp $SCRIPTS_DIR/.env $INSTALL_DIR/bin
+    error "Could not copy the .env file to the installation directory."
+}
+
+function startPowHsm() {
+    systemctl start $SERVICE_NAME
+	error "Failed to start the service."
+}
+
+function createHsmUser() {
+    if ! id -u hsm >/dev/null 2>&1; then
```

Comment:
We should probably rename the user to `powhsm`. Also, do you think it would make sense aborting if the user already exists? Why would that username exist?

Reply:
Renaming the user sounds good.

Reg. aborting, the use case I'm thinking of is the re-onboarding. If, for some reason, the user needs to use these scripts in a machine that already hosts the service, we will need a checklist of what needs to be removed. So far, we only have two items: stop / remove the service and now the user. But even then, it starts to sound like we need an `uninstall` script that performs these steps in case the user knows what he's doing, what do you think?

Maybe that's a separate task. IDK. In any case I'll rename to `powhsm` as you suggested

Reply:
Yeah, sounds like we probably need an `uninstall` script of some sort. Let's abort if the user exists (if the use case is a re-onboarding, then the service will probably also exist and the installation would abort before reaching the user creation section anyway).